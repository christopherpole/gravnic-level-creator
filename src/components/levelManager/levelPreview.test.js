import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import LevelPreview from './levelPreview';

configure({ adapter: new Adapter() });

describe('The level preview', () => {
  let props;

  beforeEach(() => {
    props = {
      tiles: [
        {
          selectedTileId: 1,
          position: 0,
        },
        {
          selectedTileId: 2,
          position: 1,
        },
        {
          selectedTileId: 3,
          position: 2,
        },
      ],
    };
  });

  it('Renders without exploding', () => {
    const level = shallow(<LevelPreview {...props} />);

    expect(level).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const level = shallow(<LevelPreview {...props} />);

    expect(toJson(level)).toMatchSnapshot();
  });
});
