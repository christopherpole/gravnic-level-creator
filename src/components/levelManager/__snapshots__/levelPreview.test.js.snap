// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The level preview Matches the current snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      key="0"
    >
      <Tile
        id={2}
      />
    </styled.div>
    <styled.div
      key="1"
    >
      <Tile
        id={3}
      />
    </styled.div>
    <styled.div
      key="2"
    >
      <Tile
        id={0}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
